### PR TITLE
connect Trustline modal to service

### PR DESCRIPTION
**Pull request by songgao**
* [x] make sure RPC types are correct after rebase

### CODE REVIEW

File: shared/wallets/trustline/asset.tsx [hunk 1]
```diff
@@ -88,6 +93,8 @@ const Asset = (props: Props) => (
               small={true}
               label="Remove"
               onClick={stopPropagation(props.onRemove)}
+              disabled={props.waitingRefresh}
+              waiting={props.waitingDelete}
             />
           ) : (
             <Kb.Button
```
Comment (chrisnojima) [inline shared/wallets/trustline/asset.tsx]: why not just use a waitingbutton

File: shared/wallets/trustline/index.tsx [hunk 2]
```diff
@@ -58,46 +68,96 @@ const sectionHeader = section =>
...
+      )}
       renderSectionHeader={({section}) => sectionHeader(section)}
     />
-  </Kb.Box2>
-)
+  )
+
+const Body = (props: BodyProps) => {
+  React.useEffect(() => props.refresh(), [])
+  React.useEffect(() => () => props.clearTrustlineModal(), [])
```
Comment (chrisnojima) [inline shared/wallets/trustline/index.tsx]: maybe these two should be merged?

File: shared/wallets/trustline/index.tsx [hunk 3]
```diff
@@ -3,100 +3,178 @@ import * as React from 'react'
...
-    <Kb.Divider />
-    {!!props.errorMessage && <Kb.Banner color="red" text={props.errorMessage} />}
+const haveEnoughBalanceToAccept = memoize(
+  (props: BodyProps) => props.balanceAvailableToSend >= Constants.trustlineHoldingBalance
+)
+
+const ListUpdateOnMount = (props: BodyProps) => {
+  // hack to get `ReactList` to render more than one item on initial mount.
+  // Somehow we need two updates here.
+  const [updateCounter, setUpdateCounter] = React.useState(0)
```
Comment (chrisnojima) [inline shared/wallets/trustline/index.tsx]: please make a ticket to remove this later. i don't think this is necessary.

Comment (songgao) [reply shared/wallets/trustline/index.tsx]: make HOTPOT-197

**Pull request merged**